src/JSObjectProxy.cc

```diff
@@ -320,6 +320,13 @@ PyObject *JSObjectProxyMethodDefinitions::JSObjectProxy_iter_next(JSObjectProxy
 
   PyObject *retVal = JSFunctionProxyMethodDefinitions::JSFunctionProxy_call(nextFunction, PyTuple_New(0), NULL);
   Py_DECREF(nextFunction);
+  if (retVal == NULL) {
+    if (PyErr_Occurred()) {
+      PyErr_PrintEx(0);    
+    }
+    PyErr_SetNone(PyExc_StopIteration);
+    return NULL;
+  }
```

Comment:
```suggestion
  if (retVal == NULL) {
    return NULL;
  }
```

This `PyErr_PrintEx(0)` is causing the stacktrace to be printed out of order, and occurs even if the exception is caught. If an exception occurs, just allow it to propagate.

Reply:
We also should not be raising a StopIteration exception here, the exception is not that the iterator has reached its end, its that this is not even an iterator to begin with